recipes/arrow/all/patches/19.0.1-0001-fix-cmake.patch

```diff
@@ -0,0 +1,75 @@
+diff --git a/cpp/cmake_modules/FindThriftAlt.cmake b/cpp/cmake_modules/FindThriftAlt.cmake
+index 98a706d..edf195e 100644
+--- a/cpp/cmake_modules/FindThriftAlt.cmake
++++ b/cpp/cmake_modules/FindThriftAlt.cmake
+@@ -45,22 +45,20 @@ endif()
+ #   * https://github.com/apache/thrift/pull/2725
+ #   * https://github.com/apache/thrift/pull/2726
+ #   * https://github.com/conda-forge/thrift-cpp-feedstock/issues/68
+-if(NOT WIN32)
+-  set(find_package_args "")
+-  if(ThriftAlt_FIND_VERSION)
+-    list(APPEND find_package_args ${ThriftAlt_FIND_VERSION})
+-  endif()
+-  if(ThriftAlt_FIND_QUIETLY)
+-    list(APPEND find_package_args QUIET)
+-  endif()
+-  find_package(Thrift ${find_package_args})
+-  if(Thrift_FOUND)
+-    set(ThriftAlt_FOUND TRUE)
+-    add_executable(thrift::compiler IMPORTED)
+-    set_target_properties(thrift::compiler PROPERTIES IMPORTED_LOCATION
+-                                                      "${THRIFT_COMPILER}")
+-    return()
+-  endif()
++set(find_package_args "")
++if(ThriftAlt_FIND_VERSION)
++  list(APPEND find_package_args ${ThriftAlt_FIND_VERSION})
++endif()
++if(ThriftAlt_FIND_QUIETLY)
++  list(APPEND find_package_args QUIET)
++endif()
++find_package(Thrift ${find_package_args})
++if(Thrift_FOUND)
++  set(ThriftAlt_FOUND TRUE)
++  add_executable(thrift::compiler IMPORTED)
++  set_target_properties(thrift::compiler PROPERTIES IMPORTED_LOCATION
++                                                    "${THRIFT_COMPILER}")
++  return()
+ endif()
+ 
+ function(extract_thrift_version)
+diff --git a/cpp/src/parquet/CMakeLists.txt b/cpp/src/parquet/CMakeLists.txt
+index 83eb52248..d165fd4d2 100644
+--- a/cpp/src/parquet/CMakeLists.txt
++++ b/cpp/src/parquet/CMakeLists.txt
+@@ -261,11 +261,11 @@ if(NOT PARQUET_MINIMAL_DEPENDENCY)
+
+   # These are libraries that we will link privately with parquet_shared (as they
+   # do not need to be linked transitively by other linkers)
+-  list(APPEND PARQUET_SHARED_PRIVATE_LINK_LIBS thrift::thrift)
++  list(APPEND PARQUET_SHARED_PRIVATE_LINK_LIBS Boost::headers thrift::thrift)
+
+   # Link publicly with parquet_static (because internal users need to
+   # transitively link all dependencies)
+-  list(APPEND PARQUET_STATIC_LINK_LIBS thrift::thrift)
++  list(APPEND PARQUET_STATIC_LINK_LIBS Boost::headers thrift::thrift)
+   if(NOT THRIFT_VENDORED)
+     list(APPEND PARQUET_STATIC_INSTALL_INTERFACE_LIBS thrift::thrift)
+   endif()
```

Comment:
(I'm an Apache Arrow developer.)

Is this change still needed? I think that we don't need this after https://github.com/apache/arrow/pull/13292 .

Reply:
> To remove the patch for `cpp/src/parquet/CMakeLists.txt`

This.

We can't remove `cpp/cmake_modules/FindThriftAlt.cmake` for now... Let's track it by https://github.com/apache/arrow/issues/45566 .

Reply:
Perfect, thank you, that appears to build correctly with the patch removed.

Reply:
Thanks!

Reply:
FYI: We'll be able to remove the `cpp/cmake_modules/FindThriftAlt.cmake` patch after Apache Arrow 20.0.0.
https://github.com/apache/arrow/pull/45567 will be included in 20.0.0.

Reply:
Awesome, thanks, good to know!